elastic/kibana
# [Fleet] Show pipelines and mappings editor for input packages

## Summary

Closes #153105

Show the pipelines and mappings editor for input packages. This can be seen when editing an input package integration policy.

Previously this was hidden for input packages and any integration package which allowed datastream to be configured. I have tweaked this logic so that we check if the matching index template for a stream exists, if it does then show the editor. 
<img width="407" alt="Screenshot 2023-03-30 at 17 02 26" src="https://user-images.githubusercontent.com/3315046/228896543-192e5fc5-3dc3-4a0f-a08f-a7ff83555ef3.png">

This allows us to handle an edge case (which will become more common as input packages are rolled out) when integration package which has since been upgraded to an input package, for example:
- user installs the custom logs integration package and sets dataset to helloworld
- the `logs-helloworld` index template is not created as integration packages only create index templates on install (custom logs creates `logs-log.log`)
- there is therefore no ingest pipeline `@custom` or `@package`component template to customize, we do not show the pipeline editor.
- The user upgrades to an input package version of custom logs (coming in v2.0.0) there still isn't an index template because we do not create them on upgrade, so we still don't want to show the pipelines and mappings editor
- Just checking for input package in this scenario would mean we showed the editor incorrectly

I have opted to show the editors even if the stream is owned by another package e.g custom logs data is being sent to an nginx stream.


### Automated tests

I have added cypress tests for checking that the mappings pipelines editors are shown for input packages.

### Manual test cases

Here is version 2.0.0 of logs which is an input package: [log-2.0.0.zip](https://github.com/elastic/kibana/files/11113579/log-2.0.0.zip) it can be added to a deployment using curl:

```
curl -XPOST -H 'content-type: application/zip' -H 'kbn-xsrf: true' http://localhost:5601/mark/api/fleet/epm/packages -u elastic:changeme --data-binary @/path/to/log-2.0.0.zip
```


- Create a policy for an integration package which does not allow dataset to be configured (e.g Apache)
    - pipelines & editor should be visible when editing package policy
- Create a policy for an integration package which does allow dataset to be configured (e.g custom logs 1.6.1)
    - pipelines & mappings editor should not be shown if dataset doesn't match the one installed by the package `logs.log`
- Upgrade custom logs to the input package version (2.0.0)
    - pipelines and mappings editor still should not be shown
